AndroidSDK/src/com/leanplum/internal/LeanplumEventDataManager.java

```diff
@@ -0,0 +1,188 @@
+  private static final String KEY_ROWID = "rowid";
+
+  private static SQLiteDatabase database;
+  private static LeanplumDataBaseManager databaseHelper;
+  private static ContentValues contentValues = new ContentValues();
+
+  public static void init(Context context) {
+    if (database != null) {
+      Log.e("Database is already initialized.");
+      return;
+    }
+    File dbFile = context.getDatabasePath(DATABASE_NAME);
+    boolean needMigration = !dbFile.exists();
+
+    // Create database if needed.
+    try {
+      if (databaseHelper == null) {
+        databaseHelper = new LeanplumDataBaseManager(context, DATABASE_NAME);
+      }
+      database = databaseHelper.getWritableDatabase();
+    } catch (Throwable t) {
+      Log.e("Cannot create database.", t);
+      Util.handleException(t);
+    }
+
+    // Migrate old data from shared preferences if needed.
+    if (database != null && needMigration) {
+      try {
+        Request.migrateFromSharedPreferences();
+      } catch (Throwable t) {
+        Log.e("Cannot move old data from shared preferences to SQLite table.", t);
+        Util.handleException(t);
+      }
+    }
+  }
+
+  /**
+   * Inserts event to event table.
+   *
+   * @param event String with json of event.
+   */
+  static void insertEvent(String event) {
+    if (database == null) {
+      return;
+    }
+    contentValues.put(COLUMN_DATA, event);
+    try {
+      database.insert(EVENT_TABLE_NAME, null, contentValues);
+    } catch (Throwable t) {
+      Log.e("Unable to insert event to database.", t);
+      Util.handleException(t);
+    }
+  }
+
+  /**
+   * Gets first count events from event table.
+   *
+   * @param count Number of events.
+   * @return List of events.
+   */
+  static List<Map<String, Object>> getEvents(int count) {
+    List<Map<String, Object>> events = new ArrayList<>();
+    if (database == null) {
+      return events;
+    }
+    Cursor cursor = null;
+    try {
+      cursor = database.query(EVENT_TABLE_NAME, new String[] {COLUMN_DATA}, null, null, null,
+          null, KEY_ROWID + " ASC", "" + count);
+      while (cursor.moveToNext()) {
+        Map<String, Object> requestArgs = JsonConverter.mapFromJson(new JSONObject(
+            cursor.getString(cursor.getColumnIndex(COLUMN_DATA))));
+        events.add(requestArgs);
+      }
+    } catch (Throwable t) {
+      Log.e("Unable to get events from the table.", t);
+      Util.handleException(t);
+    } finally {
+      if (cursor != null) {
+        cursor.close();
+      }
+    }
+    return events;
+  }
+
+  /**
+   * Deletes first count elements from event table.
+   *
+   * @param count Number of event that need to be deleted.
+   */
+  static void deleteEvents(int count) {
+    if (database == null) {
+      return;
+    }
+    try {
+      database.delete(EVENT_TABLE_NAME, KEY_ROWID + " in (select " +
+          KEY_ROWID + " from " + EVENT_TABLE_NAME + " LIMIT " + count + ")", null);
```

Comment:
but you will delete when  KEY_ROWID from this selection, no matter in which order